javascript/src/main/components/BasicCourseSearch/SelectSubject.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Form } from "react-bootstrap";
+import { compareValues } from "main/utils/sortHelper"
+
+const SelectSubject = ({ subjects, _subject, setSubject}) => {
+
+    const handleSubjectOnChange = (event) => {
+        setSubject(event.target.value);
+    };
+
+    subjects.sort(compareValues("subjectCode"));
```

Comment:
Not sure if this is something we should address in this PR given the limited amount of time that we have, but in the case that the subjects list doesn't exist due to an error in the API call, this line will cause a compilation error and lead to the app being unusable.

Reply:
So Bryan and I went to great lengths to ensure that this would not happen.    There is significant error handling code, and fallbacks.   I think we can safely set that concern aside for now.